src/datarobotx/idp/llm_blueprints.py

```diff
@@ -128,13 +160,48 @@ def get_or_register_llm_blueprint_custom_model_version(
             if bp.vector_database_settings is not None
             else None
         ),
+        guard_configs,
+        **kwargs,
     )
     for cm in dr.CustomInferenceModel.list(search_for=bp_token):  # type: ignore
         if cm.description is not None and bp_token in cm.description:
             if cm.latest_version is not None:
                 return str(cm.id), str(cm.latest_version.id)
     else:
-        cm_version = bp.register_custom_model(**kwargs)
+        prompt_column_name = kwargs.pop("prompt_column_name", None)
+        target_column_name = kwargs.pop("target_column_name", None)
+        cm_version = bp.register_custom_model(
+            prompt_column_name=prompt_column_name, target_column_name=target_column_name
+        )
         cm = dr.CustomInferenceModel.get(cm_version.custom_model_id)  # type: ignore
         cm.update(description=f"Checksum: {bp_token}")
+
+        if kwargs:
+            cm_version_id = unsafe_get_or_create_custom_model_version_from_previous(
+                endpoint=endpoint, token=token, custom_model_id=cm.id, **kwargs
+            )
+            cm_version = dr.CustomModelVersion.get(  # type: ignore[attr-defined]
```

Comment:
why do we need to get the CMV here and on line 203? seems we already ahve the id and return it at the end